crates/iota-indexer/src/backfill/ingestion/jobs/tx_wrapped_or_deleted_objects.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use diesel::RunQueryDsl;
+use iota_types::full_checkpoint_content::CheckpointData;
+
+use crate::{
+    backfill::ingestion::IngestionBackfill,
+    db::{ConnectionPool, get_pool_connection},
+    errors::IndexerError,
+    models::tx_indices::StoredTxWrappedOrDeletedObject,
+    schema::tx_wrapped_or_deleted_objects,
+};
+
+pub(crate) struct TxWrappedOrDeletedObjectsBackfill;
+
+#[async_trait::async_trait]
+impl IngestionBackfill for TxWrappedOrDeletedObjectsBackfill {
+    type ProcessedType = StoredTxWrappedOrDeletedObject;
+
+    fn process_checkpoint(checkpoint: Arc<CheckpointData>) -> Vec<Self::ProcessedType> {
+        let first_tx = checkpoint.checkpoint_summary.network_total_transactions as usize
+            - checkpoint.transactions.len();
+
+        checkpoint
+            .transactions
+            .iter()
+            .enumerate()
+            .flat_map(|(i, tx)| {
+                let effects = &tx.effects;
+
+                effects
+                    .all_tombstones()
+                    .into_iter()
+                    .chain(effects.created_then_wrapped_objects())
+                    .map(move |(object_id, _)| StoredTxWrappedOrDeletedObject {
+                        tx_sequence_number: (first_tx + i) as i64,
```

Comment:
The tx enumeration logic is correct, but I would nevertheless suggest to use [CheckpointContents::enumerate_transactions](https://iotaledger.github.io/iota/iota_types/messages_checkpoint/enum.CheckpointContents.html#method.enumerate_transactions) to have a single place where the logic is maintained and referred to.

The latter is used in the primary ingestion workflow as well.